### DISCUSSION TITLE
How to limit resource usage?

### DISCUSSION THREAD

**Discussion by mgajda** [canonical/docker-snap#242] [Q&A]
The `--cpus`, `--memory` and `--kernel-memory` options are available to limit resources used by `dockerd`.
They could be edited in `docker.service` file, but I cannot find it in the config directory.
Daemon should also respect `cgroup`, but it is not clear how to set it up with Snap.

Ideally end user could limit Docker daemon to half CPUs and half memory, since otherwise root process will be scheduled above all UI processes and may hang the machine.

**Comment by mgajda** [comment]
It is possible to use `cgset` from `cgroup-utils` package:
```
sudo cgset -r cpuset.cpus=14-15 /system.slice/snap-docker-2285.mount
```

However this command would need to be updated for every new release of the Snap.

Would it be possible to name `cgroup` by Snap name, not name and version?

**Comment by locnnil** [comment]
Snaps directories have the `current` directory that is basically a symlink that points to the current revision being used:

```console
$ ls -l /snap/docker/
total 0
drwxr-xr-x 12 root root 143 Jun 25 19:40 2932
drwxr-xr-x 12 root root 143 Sep 20 13:26 2969
lrwxrwxrwx  1 root root   4 Oct  8 14:50 current -> 2932
```
I'm wondering if it would be possible to do that for `cgroups`, if so, that would be an implementation for [snapd](https://github.com/canonical/snapd). 

But first, I had to investigate if that is the most indicated way to do that.
